backend/deepchecks_monitoring/api/v1/check.py

```diff
@@ -190,10 +192,12 @@ async def delete_check_by_id(
         check_identifier: CheckIdentifier = CheckIdentifier.resolver(),
         session: AsyncSession = AsyncSessionDep,
         user: User = Depends(auth.CurrentUser()),
+        resources_provider: ResourcesProvider = ResourcesProviderDep,
 ):
     """Delete check instance by identifier."""
     model = await fetch_or_404(session, Model, **model_identifier.as_kwargs)
-    await Model.fetch_or_403(session, model.id, user)
+    if resources_provider.get_features_control(user).model_assignment:
+        await Model.fetch_or_403(session, model.id, user)
```

Comment:
why fetch the model twice? can't do it with some if?
```
if resources_provider.get_features_control(user).model_assignment:
        model = await Model.fetch_or_403(session, model.id, user)
else:
      model = await fetch_or_404(session, Model, **model_identifier.as_kwargs)
```

Reply:
feels like it can be done without doing the same query twice

Reply:
got an idea of how?

Reply:
I haven't looked what is going on inside the `fetch_or_403` but I'd change it name and have it doing something like

1. get model
2. if not found raise 404
3. check permissions
4. if not allowed 403

Reply:
the usage is different, it starts with the 3rd step.
it is also not a new code